validator/keymanager/local/delete.go

```diff
@@ -16,17 +15,27 @@ import (
 // DeleteKeystores takes in public keys and removes the accounts from the wallet.
 // This includes their disk keystore and cached keystore, but maintains the slashing
 // protection history in the database.
+// 1) Copy the in memory keystore
+// 2) Delete the keys from copied in memory keystore
+// 3) Save the copy to disk
+// 4) Reinitialize account store from disk
+// 5) Verify keys are indeed deleted
+// 6) Return API response
 func (km *Keymanager) DeleteKeystores(
 	ctx context.Context, publicKeys [][]byte,
 ) ([]*ethpbservice.DeletedKeystoreStatus, error) {
 	// Check for duplicate keys and filter them out.
 	trackedPublicKeys := make(map[[fieldparams.BLSPubkeyLength]byte]bool)
 	statuses := make([]*ethpbservice.DeletedKeystoreStatus, 0, len(publicKeys))
-	var store *AccountsKeystoreRepresentation
-	var err error
 	deletedKeys := make([][]byte, 0, len(publicKeys))
+	originalKeyLen := len(km.accountsStore.PublicKeys)
+	// 1) Copy the in memory keystore
+	storeCopy := &accountStore{}
```

Comment:
its probably better to have this as a receiver method of the `accountStore` struct